taichi/transforms/make_block_local.cpp

```diff
@@ -67,45 +68,58 @@ void make_block_local_offload(OffloadedStmt *offload,
             block = std::make_unique<Block>();
             block->parent_stmt = offload;
           }
-          Stmt *block_linear_index =
+          // Equivalent to CUDA threadIdx
+          Stmt *thread_idx_stmt =
               block->push_back<LoopLinearIndexStmt>(offload);
 
           /*
           Note that since there are fewer elements in the block than in BLS,
           each thread may have to fetch more than one element to BLS.
           Therefore on CUDA we need something like
 
-          auto bls_element_id = block_linear_index;
-          while (bls_element_id < bls_size) {
+          auto bls_element_id = thread_idx_stmt;
+          while (bls_element_id < bls_num_elements) {
             i, j, k = bls_to_global(bls_element_id)
             bls[bls_element_id] = x[i, j, k]
             // or x[i, j, k] = bls[bls_element_id]
             bls_element_id += block_dim;
           }
 
+          func bls_to_global(bls_element_id):
+            partial = bls_element_id
+            global_indices = []  // "i, j, k"
+            for i in [(dim - 1), 0]:
```

Comment:
nit:
```suggestion
            for i in reversed(range(0, dim)):
```
`[(dim - 1), 0]` itself has a different meaning in Python so I think we should avoid it, and `range(dim - 1, -1, -1)` (if we replace it with this one) is a little bit hard to understand.